aggregator_api/src/lib.rs

```diff
@@ -0,0 +1,814 @@
+        .map(|(_, v)| TaskId::from_str(v))
+        .transpose()
+        .map_err(|err| {
+            warn!(err = %err, "Couldn't parse pagination_token");
+            Status::BadRequest
+        })?;
+
+    let task_ids = ds
+        .run_tx_with_name("get_task_ids", |tx| {
+            Box::pin(async move { tx.get_task_ids(lower_bound).await })
+        })
+        .await
+        .map_err(|err| {
+            error!(err = %err, "Database transaction error");
+            Status::InternalServerError
+        })?;
+    let pagination_token = task_ids.last().cloned();
+
+    Ok((
+        Json(GetTaskIdsResp {
+            task_ids,
+            pagination_token,
+        }),
+        Halt,
+    ))
+}
+
+async fn post_task<C: Clock>(
+    _: &mut Conn,
+    (State(ds), Json(req)): (State<Arc<Datastore<C>>>, Json<PostTaskReq>),
+) -> Result<impl Handler, Status> {
+    let vdaf_verify_keys = Vec::from([SecretBytes::new(
+        thread_rng()
+            .sample_iter(Standard)
+            .take(req.vdaf.verify_key_length())
+            .collect(),
+    )]);
+    let task_expiration = Time::from_seconds_since_epoch(req.task_expiration);
+    let time_precision = Duration::from_seconds(req.time_precision);
+    let collector_auth_tokens = match req.role {
+        Role::Leader => Vec::from([random()]),
+        _ => Vec::new(),
+    };
+    let hpke_keys = Vec::from([generate_hpke_config_and_private_key(
+        random(),
+        HpkeKemId::X25519HkdfSha256,
+        HpkeKdfId::HkdfSha256,
+        HpkeAeadId::Aes128Gcm,
+    )]);
+
+    let task = Arc::new(
+        Task::new(
+            /* task_id */ random(),
+            /* aggregator_endpoints */ req.aggregator_endpoints,
+            /* query_type */ req.query_type,
+            /* vdaf */ req.vdaf,
+            /* role */ req.role,
+            /* vdaf_verify_keys */ vdaf_verify_keys,
+            /* max_batch_query_count */ req.max_batch_query_count,
+            /* task_expiration */ task_expiration,
+            /* report_expiry_age */
+            Some(Duration::from_seconds(3600 * 24 * 7 * 2)), // 2 weeks
+            /* min_batch_size */ req.min_batch_size,
+            /* time_precision */ time_precision,
+            /* tolerable_clock_skew */
+            Duration::from_seconds(60), // 1 minute,
+            /* collector_hpke_config */ req.collector_hpke_config,
+            /* aggregator_auth_tokens */ Vec::from([random()]),
+            /* collector_auth_tokens */ collector_auth_tokens,
+            /* hpke_keys */ hpke_keys,
+        )
+        .map_err(|_| Status::BadRequest)?,
+    );
+
+    ds.run_tx_with_name("post_task", |tx| {
+        let task = Arc::clone(&task);
+        Box::pin(async move { tx.put_task(&task).await })
+    })
+    .await
+    .map_err(|err| {
+        error!(err = %err, "Database transaction error");
+        Status::InternalServerError
+    })?;
+
+    Ok((Json(TaskResp::from(task.as_ref())), Halt))
```

Comment:
It's not necessary to halt in a terminal handler, so this could just be `Ok(Json(TaskResp::from(task.as_ref())))`